src/components/document/document.tsx

```diff
@@ -96,6 +96,12 @@ const DeleteButton = ({ onClick }: { onClick: () => void }) => {
   );
 };
 
+const DeleteDisabledButton = () => {
```

Comment:
Instead of a separate component, make `enabled` (or `disabled` if the logic's easier) a property of the `DeleteButton` component.